docs/site/submitting_a_pr.md

```diff
@@ -0,0 +1,495 @@
+
+To help with abiding by the rules of commit messages, please use the
+`commitizen` tool mentioned in the documentation above. This means will we use
+`git cz` instead of `git commit`. Install `commitizen` by running
+`npm install -g commitizen`.
+
+In my case, I want the commit message to look like this:
+
+```
+docs: tutorial on how to submit a pull request to lb4
+
+This tutorial shows contributors step-by-step instructions on how to submit a pull request (PR) to LoopBack v4
+
+```
+
+To start `commitizen`, run :
+
+```
+git cz
+```
+
+Follow the prompts for the various fields.
+
+In my case I have entered :
+
+```
+type                : docs
+scope               : <blank>
+short description   : tutorial on how to submit a pull request to lb4
+long description    : This tutorial shows contributors step-by-step instructions on how to submit a pull request (PR) to LoopBack v4
+breaking changes    : N
+affects open issues : N
+```
+
+![submit_pr_git_cz_1.png](./imgs/submit_pr_git_cz_1.png)
+
+The interactive `commitizen` prompts complete and the commit is created with a
+properly formatted message.
+
+#### 5. Push commits to the remote feature branch
+
+It is now time to push your local committed changes of your local feature branch
+directory to your remote feature branch; to keep them in synch.
+
+The very `first` time you push your changes you should run:
+
+```
+git push --set-upstream origin emonddr-doc-changes
+```
+
+The `--set-upstream` parameter will set up the tracking between your local
+feature branch and your remote feature branch.
+
+Afterwards, for any additional push, you just need to run:
+
+```
+git push
+```
+
+It is not necessary to create a pull request immediately on the push of your
+first commit; this can be done later.
+
+Do take some time to think about how many commits you want in the pull request
+when you eventually create it.
+
+If it makes sense to place your changes into one commit, then do so.
+
+If, however, it makes more sense to place your changes into a few logically
+separate commits, then do so.
+
+Whatever is the best way for the maintainers to understand the changes when the
+pull request is being reviewed.
+
+You can always re-organize commits or compress the number of commits later
+through a step called `Squashing Commits`.
+
+#### 6. Rebasing
+
+Eventually your fork of the original repository will become stale, and it will
+be necessary to bring it up-to-date.
+
+It is recommended that you perform a `rebase` before actually creating a pull
+request; if necessary.
+
+In my case, the fork of the repository is behind by several commits.
+
+![submit_pr_rebase_1.png](./imgs/submit_pr_rebase_1.png)
+
+It is necessary to perform a `rebase` .
+
+To `rebase` your forked repository's `master` branch off of the original
+repository, run :
+
+```
+git remote add upstream git@github.com:strongloop/loopback-next.git
+git checkout master
+git pull --rebase upstream master
```

Comment:
Does this rebase command go into an interactive rebase?

Reply:
not it just pulls everything down.